src/com/fundation/search/model/Search.java

```diff
@@ -222,13 +435,29 @@ public void initSearch() {
     public List<FileResult> getResultList() {
         List<FileResult> result = new ArrayList<>();
         if (!fileList.isEmpty()) {
+            for (File file : fileList) {
+                try {
+                    BasicFileAttributes fileAttributes = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+                    DosFileAttributes fileAttributes1 = Files.readAttributes(file.toPath(), DosFileAttributes.class);
```

Comment:
You are apply Builder here, but maybe it is mor better on  "private void filterByCriteria(SearchCriteria criteria)" Method, try improve this.